internal/bminventory/inventory_v2_handlers.go

```diff
@@ -638,18 +693,33 @@ func (b *bareMetalInventory) infraEnvIPXEScript(ctx context.Context, infraEnv *c
 		return "", errors.Wrap(err, "failed to sign initrd URL")
 	}
 
-	return fmt.Sprintf(ipxeScriptFormat, initrdURL, kernelURL, rootfsURL), nil
+	return fmt.Sprintf(ipxeBootScriptFormat, initrdURL, kernelURL, rootfsURL), nil
+}
+
+func (b *bareMetalInventory) infraEnvIPXEScript(ctx context.Context, infraEnv *common.InfraEnv, mac *strfmt.MAC, bootControl bool) (string, error) {
+	if mac != nil && *mac != "" {
+		if err := b.canServeHostIPXEScript(infraEnv, mac); err != nil {
+			return "", err
+		}
+	} else if bootControl {
+		return b.hostRedirectIPXEScript(ctx, infraEnv)
+	}
+	return b.bootIPXEScript(ctx, infraEnv)
 }
 
 func (b *bareMetalInventory) GetInfraEnvPresignedFileURL(ctx context.Context, params installer.GetInfraEnvPresignedFileURLParams) middleware.Responder {
+	if swag.BoolValue(params.BootControl) && params.FileName != "ipxe-script" {
+		return common.NewApiError(http.StatusBadRequest, errors.New(`boot_control can be set only for "ipxe-script"`))
+	}
 	infraEnv, err := common.GetInfraEnvFromDB(b.db, params.InfraEnvID)
 	if err != nil {
 		return common.GenerateErrorResponder(err)
 	}
 
 	builder := &installer.V2DownloadInfraEnvFilesURL{
-		InfraEnvID: params.InfraEnvID,
-		FileName:   params.FileName,
+		InfraEnvID:  params.InfraEnvID,
+		FileName:    params.FileName,
+		BootControl: params.BootControl,
```

Comment:
Will this lead to a query parameter like `boot_control=` with no value?

Reply:
Checking it in unit test